nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/grok/GrokReader.java

```diff
@@ -0,0 +1,111 @@
+    + "If a line in the input does not match the expected message pattern, the line of text is considered to be part of the previous "
+    + "message, with the exception of stack traces. A stack trace that is found at the end of a log message is considered to be part "
+    + "of the previous message but is added to the 'STACK_TRACE' field of the Record. If a record has no stack trace, it will have a NULL value "
+    + "for the STACK_TRACE field. All fields that are parsed are considered to be of type String by default. If there is need to change the type of a field, "
+    + "this can be accomplished by configuring the Schema Registry to use and adding the appropriate schema.")
+public class GrokReader extends SchemaRegistryRecordReader implements RowRecordReaderFactory {
+    private volatile Grok grok;
+    private volatile boolean useSchemaRegistry;
+
+    private static final String DEFAULT_PATTERN_NAME = "/default-grok-patterns.txt";
+
+    static final PropertyDescriptor PATTERN_FILE = new PropertyDescriptor.Builder()
+        .name("Grok Pattern File")
+        .description("Path to a file that contains Grok Patterns to use for parsing logs. If not specified, a built-in default Pattern file "
+            + "will be used. If specified, all patterns in the given pattern file will override the default patterns. See the Controller Service's "
+            + "Additional Details for a list of pre-defined patterns.")
+        .addValidator(StandardValidators.FILE_EXISTS_VALIDATOR)
+        .expressionLanguageSupported(true)
+        .required(false)
+        .build();
+
+    static final PropertyDescriptor GROK_EXPRESSION = new PropertyDescriptor.Builder()
+        .name("Grok Expression")
+        .description("Specifies the format of a log line in Grok format. This allows the Record Reader to understand how to parse each log line. "
+            + "If a line in the log file does not match this pattern, the line will be assumed to belong to the previous log message.")
+        .addValidator(new GrokExpressionValidator())
+        .required(true)
+        .build();
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = new ArrayList<>(super.getSupportedPropertyDescriptors());
+        properties.add(PATTERN_FILE);
+        properties.add(GROK_EXPRESSION);
+        return properties;
+    }
+
+    @OnEnabled
+    public void preCompile(final ConfigurationContext context) throws GrokException, IOException {
+        grok = new Grok();
+
+        try (final InputStream in = getClass().getResourceAsStream(DEFAULT_PATTERN_NAME);
+            final Reader reader = new InputStreamReader(in)) {
+            grok.addPatternFromReader(reader);
+        }
+
+        if (context.getProperty(PATTERN_FILE).isSet()) {
+            grok.addPatternFromFile(context.getProperty(PATTERN_FILE).getValue());
```

Comment:
Since it appears to be that EL is supported should it evaluate the expression and not just get value?